### PR TITLE
233 drop python 38 support

### PR DESCRIPTION
**Pull request by waltsims**
This PR drops Python 3.8 support from k-wave-python

### CODE REVIEW

File: .github/workflows/pytest.yml [hunk 1]
```diff
@@ -47,7 +47,7 @@ jobs:
     strategy:
       matrix:
         os: [ "windows-latest", "ubuntu-latest" ] #, "macos-latest"]
-        python-version: [ "3.8", "3.9", "3.10" ]
+        python-version: [ "3.9", "3.10" ]
```
Comment (faridyagubbayli) [inline .github/workflows/pytest.yml:50]: What about extending the scope of the PR and adding 3.11 support here?

**Comment by faridyagubbayli** [comment]
I'd love to see us dropping 3.8 :) There are many useful syntax magics in 3.9 that we can't usually use due to 3.8 support.

File: .github/workflows/pytest.yml (hunk 1 above)
Comment (waltsims) [reply .github/workflows/pytest.yml:50]: We have [this PR](https://github.com/waltsims/k-wave-python/pull/206) for that. It is currently failing due to the uff dependency. What will be alleviated with https://github.com/waltsims/k-wave-python/pull/229

Comment (faridyagubbayli) [reply .github/workflows/pytest.yml]: Both PRs are merged now. I updated the branch with `master` and resolved the conflicts. Maybe it is time to merge? :)

Comment (faridyagubbayli) [reply .github/workflows/pytest.yml]: And tested passed

Comment (waltsims) [reply .github/workflows/pytest.yml]: Pre or post release? We won't make any python 3.8 braking changes yet will we?

Comment (faridyagubbayli) [reply .github/workflows/pytest.yml]: Good point, no breaking changes on my side. We can do it post release also.

**Pull request merged**